program/rust/src/rust_oracle.rs

```diff
@@ -803,3 +836,77 @@ pub fn del_product(
 
     Ok(())
 }
+
+
+pub fn upd_permissions(
+    program_id: &Pubkey,
+    accounts: &[AccountInfo],
+    instruction_data: &[u8],
+) -> ProgramResult {
+    let [funding_account, program_account, programdata_account, permissions_account, system_program] =
+        match accounts {
+            [v, w, x, y, z] => Ok([v, w, x, y, z]),
+            _ => Err(ProgramError::InvalidArgument),
+        }?;
+
+    let cmd_args = load::<UpdPermissionsArgs>(instruction_data)?;
+
+    check_valid_funding_account(funding_account)?;
+    check_is_valid_upgradeauthority_program_programdata_triplet(
+        funding_account,
+        program_account,
+        programdata_account,
+        program_id,
+    )?;
+    let bump = check_pda(
+        permissions_account,
+        &[PERMISSIONS_SEED.as_bytes()],
+        program_id,
+    )?;
+    pyth_assert(
```

Comment:
`check_pda` should not return a `bump` - could we move the `find_program_address` into the top-level `upd_permissions` function, so that `check_pda` just becomes a thin wrapper around the `pyth_assert`, like the other `check_*` functions?

Reply:
Yes